src/discover/learner.js

```diff
@@ -0,0 +1,28 @@
+const {existsSync} = require('fs')
+const Learner = require('ac-learn')
```

Comment:
I'm -1 on directly including a machine learning approach such as `ac-learn` for a couple reasons:

* It's not 100% predictable, only high 90s at best. For dev tooling I personally prefer simpler, predictable patterns that can be debugged & changed more directly & transparently
* It requires more compute & storage per-repository (see the large `learner.json` file here)

It feels ... unusual? to me to start off by including the more heavyweight, targeted approach directly in the CLI (or at least its repository).

From my (admittedly not very informed) perspective, it would be nice to see a bit more trying it out before fully onboarding. Could we add it to the docs first as a third party approach? See existing repos try it out successfully?

By the way, sorry for not posting serious thoughts on this till now 😞. I'd been meaning to say something and it slipped my mind. But I'm very up for discussing more - and am not very high confidence that what I'm saying is at all reasonable!

Reply:
I see where you're coming from and initially started that feature with some non-ML pattern matching and stuff, which were... okay but not scalable enough to support enough of the contributions. 
I guess the ML-driven categorisation could be put behind a command-line flag (like a feature flag) and when it's off it will only categorise the minimal set of contribution types (like review, code, bug and docs).